### PR TITLE
internal/plugins/manifests: inject operator image using `IMG=<tag>`

### PR DESCRIPTION
**Pull request by estroz**
**Description of the change:**
* internal/plugins/manifests: add `IMG` variable injection to `bundle` recipe
* test/: update tests to use `IMG=<test image tag>`
* website/: update OLM integration docs with this change

**Motivation for the change:** This PR adds `cd config/manager && $(KUSTOMIZE) edit set image controller=$(IMG)` to the manifests phase 2 plugin's Makefile modifier, such that a user can specify operator image using `make bundle IMG=<tag>`. Our e2e tests were passing because `make deploy IMG=<tag>` was run beforehand; in a newly scaffolded project that only uses bundles/package manifests, that will likely not be the case.

/cc @joelanford @camilamacedo86 @asmacdo @fabianvf 

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [x] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [x] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

### CODE REVIEW

File: test/e2e-helm/e2e_helm_olm_test.go [hunk 1]
```diff
@@ -50,12 +49,7 @@ var _ = Describe("Integrating Helm Projects with OLM", func() {
 
 		It("should generate and run a valid OLM bundle and packagemanifests", func() {
 			By("building the bundle")
-			err := tc.Make("bundle")
-			Expect(err).NotTo(HaveOccurred())
-
-			By("validating the bundle")
-			bundleValidateCmd := exec.Command(tc.BinaryName, "bundle", "validate", "bundle")
-			_, err = tc.Run(bundleValidateCmd)
```
Comment (joelanford) [inline test/e2e-helm/e2e_helm_olm_test.go:58]: Why is this removed?

File: test/e2e-helm/e2e_helm_olm_test.go [hunk 2]
```diff
@@ -77,17 +71,7 @@ var _ = Describe("Integrating Helm Projects with OLM", func() {
...
-			Expect(err).NotTo(HaveOccurred())
-
-			By("updating clusterserviceversion with the manager image")
-			testutils.ReplaceInFile(
-				filepath.Join(tc.Dir, "packagemanifests", operatorVersion,
-					fmt.Sprintf("e2e-%s.clusterserviceversion.yaml", tc.TestSuffix)),
-				"controller:latest", tc.ImageName)
-
-			By("installing crds to run packagemanifests")
-			err = tc.Make("install")
```
Comment (joelanford) [inline test/e2e-helm/e2e_helm_olm_test.go:90]: +1

File: test/e2e-helm/e2e_helm_olm_test.go (hunk 1 above)
Comment (estroz) [reply test/e2e-helm/e2e_helm_olm_test.go:58]: `bundle validate` is already called by `make bundle`.

File: test/e2e-helm/e2e_helm_olm_test.go (hunk 2 above)
Comment (estroz) [reply test/e2e-helm/e2e_helm_olm_test.go:90]: OLM should be installing these in `run packagemanifests`.

Comment (joelanford) [reply test/e2e-helm/e2e_helm_olm_test.go:90]: Yep!
